src/EnergyPlus/Pipes.cc

```diff
@@ -134,130 +135,49 @@ namespace Pipes {
 		LocalPipe.deallocate();
 	}
 
-	void
-	SimPipes(
-		int const CompType,
-		std::string & PipeName,
-		int & CompIndex,
-		Real64 const EP_UNUSED( MaxVolFlowRate ),
-		bool const EP_UNUSED( InitLoopEquip ),
-		bool const EP_UNUSED( FirstHVACIteration )
-	)
-	{
-		// SUBROUTINE INFORMATION:
-		//       AUTHOR         Dan Fisher
-		//       DATE WRITTEN   Sept. 1998
-		//       MODIFIED       na
-		//       RE-ENGINEERED  na
-
-		// PURPOSE OF THIS SUBROUTINE:
-
-		// METHODOLOGY EMPLOYED:
-
-		// REFERENCES:
-
-		// Using/Aliasing
-		using InputProcessor::FindItemInList;
-		using InputProcessor::SameString;
-		using General::TrimSigDigits;
-		using DataPlant::PlantLoop;
-		using DataPlant::ScanPlantLoopsForObject;
-		using PlantUtilities::SafeCopyPlantNode;
-		using PlantUtilities::InitComponentNodes;
-		using DataGlobals::BeginEnvrnFlag;
-
-		// Locals
-		// SUBROUTINE PARAMETER DEFINITIONS:
-
-		// SUBROUTINE ARGUMENT DEFINITIONS:
-
-		// INTERFACE BLOCK SPECIFICATIONS
-
-		// DERIVED TYPE DEFINITIONS
-
-		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-		int PipeNum;
-		//  INTEGER :: LoopNum
-		//  INTEGER :: LoopSideNum
-		//  INTEGER :: BranchNum
-		//  INTEGER :: CompNum
-		int FoundOnLoop;
-		bool errFlag;
-
-		//FLOW
-
-		//GET INPUT
+	std::shared_ptr< PlantComponent > LocalPipeData::factory( int objectType, std::string objectName ) {
+		// Process the input data for pipes if it hasn't been done already
 		if ( GetPipeInputFlag ) {
```

Comment:
That sounds good.  But it seems like it is one step farther than I would like to take this initial pass.  I'm pushing to keep it like this for right now, and then proceed further with improvements as you described in a later refactoring.  The main reason is because when we try to scoot through _all_ the plant components, with the current way we don't have to modify GetInput...at all.  And that will really speed this process up.